general/ordering.cpp

```diff
@@ -0,0 +1,67 @@
+#include "ordering.hpp"
+
+namespace mfem
+{
+
+template <>
+void Ordering::DofsToVDofs<Ordering::byNODES>(int ndofs, int vdim,
+                                              Array<int> &dofs)
+{
+   // static method
+   int size = dofs.Size();
+   dofs.SetSize(size*vdim);
+   for (int vd = 1; vd < vdim; vd++)
+   {
+      for (int i = 0; i < size; i++)
+      {
+         dofs[i+size*vd] = Map<byNODES>(ndofs, vdim, dofs[i], vd);
+      }
+   }
+}
+
+template <>
+void Ordering::DofsToVDofs<Ordering::byVDIM>(int ndofs, int vdim,
+                                             Array<int> &dofs)
+{
+   // static method
+   int size = dofs.Size();
+   dofs.SetSize(size*vdim);
+   for (int vd = vdim-1; vd >= 0; vd--)
+   {
+      for (int i = 0; i < size; i++)
+      {
+         dofs[i+size*vd] = Map<byVDIM>(ndofs, vdim, dofs[i], vd);
+      }
+   }
+}
+
+void Ordering::Reorder(Vector &v, int vdim, Ordering::Type in_ord,
+                       Ordering::Type out_ord)
+{
+   if (in_ord == out_ord)
+   {
+      return;
+   }
+
+   int nvdofs = v.Size();
+   int nldofs = nvdofs/vdim;
+
+   if (out_ord == Ordering::byNODES) // byVDIM -> byNODES
+   {
+      Vector temp = v;
+      for (int i = 0; i < nvdofs; i++)
+      {
+         v[i] = temp[Map<byVDIM>(nldofs,vdim,i%nldofs,i/nldofs)];
```

Comment:
Put the index on a different line to make it more readable and also I would suggest replacing the division and modulo by nested `for` loop. If the stride is not a power of two and the compiler does not optimize it out, division/modulo are unnecessarily expensive operations compared to multiplication, which is usually handled by address calculation instructions.